packages/credential-sdk/src/types/did/document/service-endpoint.js

```diff
@@ -133,34 +138,121 @@ export class ServiceEndpointOrigins extends TypedArray {
   static Class = ServiceEndpointOrigin;
 }
 
+export class DIDCommServiceEndpoint extends TypedStruct {
+  static Classes = {
+    uri: TypedString,
+    accept: class AcceptArray extends TypedArray {
+      static Class = TypedString;
+    },
+    routingKeys: class RoutingKeysArray extends TypedArray {
+      static Class = TypedString;
+    },
+    recipientKeys: class RecipientKeysArray extends TypedArray {
+      static Class = TypedString;
+    },
+    priority: class PriorityNumber extends TypedNumber { },
+  };
+
+  constructor(uri, accept = [], routingKeys = [], recipientKeys = [], priority = 0) {
+    super(uri, accept, routingKeys, recipientKeys, priority);
```

Comment:
should we have some object validation here? i.e. ensure URI is actually valid

for priority, are we assuming a higher number is a higher priority?